samples/net/sockets/echo_server/boards/qemu_cortex_a53-e1000.overlay

```diff
@@ -0,0 +1,16 @@
+&pcie {
+	eth0: eth@0 {
+		compatible = "intel,e1000";
+
+		vendor-id = <0x8086>;
+		device-id = <0x100e>;
+
+		interrupts = <2>;
+		interrupt-parent = <&pcie>;
+
+		/* Prevent device tree error "lacks 'reg' property" */
+		reg = <0x00 0x00 0x00 0x00 0x01>;
```

Comment:
Without this dummy `reg` node there is an error:

> devicetree error: <Node /soc/pcie@4010000000/eth@0 in '/path/to/zephyr/misc/empty_file.c'> lacks 'reg' property (needed for 'interrupt-map' unit address lookup)

Maybe there is a better way to do this?

Reply:
From what I can tell this should actually look like this:

	eth0: pcie@1,0 {                    // <-- edtlib.py: pcie@1,0 ..., binding ... intel,e1000.yaml has non-hex unit address
		compatible = "intel,e1000";
		vendor-id = <0x8086>;
		device-id = <0x100e>;
		interrupts = <1>;
		interrupt-parent = <&pcie>;
		reg = <0x0800 0 0 0 0>;     // <-- edtlib.py: zero-sized 'reg' in ... seems meaningless
		                            // <-- edtlib.py: unit address and first address in 'reg' (0x8000000000000000000) don't match for /soc/pcie@4010000000/pcie@800
		status = "okay";
	};

Setting `reg = <0x0800` automatically handles the INTx "swizzling" so `interrupts = <1>;` can be used (instead of `<2>`).

This is more like what is done in Linux, however edtlib.py currently raises a few errors and might need a biggish update to enable this

Reply:
Turns out this wasn't too difficult to get working in edtlib.py so I made those changes